File: LeetCode_scripts/python/two-sum/two_sum_test.py
import unittest

from two_sum import Solution

class TestTwoSum(unittest.TestCase):
    def test_sum(self):
        """
        Given nums = [2, 7, 11, 15], target = 9,
        Because nums[0] + nums[1] = 2 + 7 = 9,
        return [0, 1].
        """

        nums = [2, 7, 11, 15]
        target = 9
        expected = [0, 1]

        sol = Solution()
        result = sol.twoSum(nums, target)
        self.assertListEqual(result, expected)

if __name__ == '__main__':
    unittest.main()

File: LeetCode_scripts/go/Two-Sum/twoSum_test.go
package twosum

import (
	"reflect"
	"testing"
)

func TestTwoSum(t *testing.T) {
	tt := []struct {
		name   string
		slice  []int
		target int
		result []int
	}{
		{"primul", []int{2, 7, 11, 15}, 9, []int{0, 1}},
		{"urmatorul", []int{5, 7, 3, 15}, 8, []int{0, 2}},
	}

	for _, tc := range tt {
		t.Run(tc.name, func(t *testing.T) {
			result := twoSum(tc.slice, tc.target)
			if !reflect.DeepEqual(tc.result, result) {
				t.Errorf("expected result to be %v; got %v", tc.result, result)
			}
		})
	}
}
